subsystem/kubeapi_test.go

```diff
@@ -0,0 +1,305 @@
+	Expect(cluster.ServiceNetworkCidr).Should(Equal(spec.ServiceNetworkCidr))
+	Expect(cluster.IngressVip).Should(Equal(spec.IngressVip))
+	Expect(cluster.SSHPublicKey).Should(Equal(spec.SSHPublicKey))
+	Expect(swag.BoolValue(cluster.VipDhcpAllocation)).Should(Equal(spec.VIPDhcpAllocation))
+	Expect(cluster.HTTPProxy).Should(Equal(spec.HTTPProxy))
+	Expect(cluster.HTTPSProxy).Should(Equal(spec.HTTPSProxy))
+	Expect(cluster.NoProxy).Should(Equal(spec.NoProxy))
+	Expect(swag.BoolValue(cluster.UserManagedNetworking)).Should(Equal(spec.UserManagedNetworking))
+	Expect(cluster.OpenshiftVersion).Should(Equal(spec.OpenshiftVersion))
+	Expect(cluster.AdditionalNtpSource).Should(Equal(spec.AdditionalNtpSource))
+	Expect(cluster.PullSecret).Should(Equal(pullSecret))
+}
+
+func deployImageCRD(ctx context.Context, client k8sclient.Client, spec *v1alpha1.ImageSpec) {
+	err := client.Create(ctx, &v1alpha1.Image{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Image",
+			APIVersion: getAPIVersion(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: Options.Namespace,
+			Name:      spec.ClusterRef.Name,
+		},
+		Spec: *spec,
+	})
+	Expect(err).ShouldNot(HaveOccurred())
+}
+
+func waitForImageCRDState(
+	ctx context.Context, client k8sclient.Client, key types.NamespacedName, message string, timeout int) {
+
+	image := &v1alpha1.Image{}
+	start := time.Now()
+	for time.Duration(timeout)*time.Second > time.Since(start) {
+		image = getImageCRD(ctx, client, key)
+		if image.Status.State != "" {
+			Expect(image.Status.State).Should(Equal(message))
+		}
+		time.Sleep(time.Second)
+	}
+	Expect(image.Status.State).Should(Equal(message))
+}
+
+func getImageCRD(ctx context.Context, client k8sclient.Client, key types.NamespacedName) *v1alpha1.Image {
+	image := &v1alpha1.Image{}
+	err := client.Get(ctx, key, image)
+	Expect(err).ShouldNot(HaveOccurred())
+	return image
+}
+
+func cleanUP(ctx context.Context, client k8sclient.Client) {
+	Expect(client.DeleteAllOf(ctx, &v1alpha1.Cluster{}, k8sclient.InNamespace(Options.Namespace))).ShouldNot(HaveOccurred())
+	Expect(client.DeleteAllOf(ctx, &v1alpha1.Image{}, k8sclient.InNamespace(Options.Namespace))).ShouldNot(HaveOccurred())
+}
+
+var _ = Describe("cluster installation", func() {
+	ctx := context.Background()
+
+	waitForClusterReconcileTimeout := 10
+
+	secretRef := &corev1.SecretReference{}
+
+	generateFAPostStepReply := func(h *models.Host, freeAddresses models.FreeNetworksAddresses) {
+		fa, err := json.Marshal(&freeAddresses)
+		Expect(err).NotTo(HaveOccurred())
+		_, err = agentBMClient.Installer.PostStepReply(ctx, &installer.PostStepReplyParams{
+			ClusterID: h.ClusterID,
+			HostID:    *h.ID,
+			Reply: &models.StepReply{
+				ExitCode: 0,
+				Output:   string(fa),
+				StepID:   string(models.StepTypeFreeNetworkAddresses),
+				StepType: models.StepTypeFreeNetworkAddresses,
+			},
+		})
+		Expect(err).ShouldNot(HaveOccurred())
+	}
+
+	BeforeSuite(func() {
+		if !Options.EnableKubeAPI {
```

Comment:
it will be resolved with specific marking for the tests.

Reply:
+1